File: src/main/java/hmm/Flag.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package hmm;

import java.io.IOException;
import java.util.Properties;

/**
 *
 * @author ray
 */
public class Flag {

    private final Properties p = new Properties();
    private final static Flag flag = new Flag();
    public final int n = 3;
    private final double[] labda = new double[n];
    private final int ngramMinimumShowTimes = Integer.parseInt(p.getProperty("NgramMinimumShowTimes", "1"));

    Flag() {
        labda[0] = 1.0 / 6.0;
        labda[1] = 1.0 / 3.0;
        labda[2] = 1.0 / 2.0;

        try {
            p.load(Flag.class.getClassLoader().getResourceAsStream("pinyin.properties"));
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }
    }

    public static Flag getInstance() {
        return flag;
    }

    public int getNgramMinimumShowTimes() {
        return ngramMinimumShowTimes;
    }

    public double labda(int i) {
        return labda[i];
    }
}

File: lib.hmm/src/main/scala/hmm/Flag.scala
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package hmm

import java.util.Properties

object Flag {

  val n = 3;
  val p = new Properties();
  val labda: Array[Double] = new Array[Double](n)
  var ngramMinimumShowTimes = 1;

  {
    try {
      p.load(Flag.getClass.getClassLoader().getResourceAsStream("pinyin.properties"));
    } catch  {
      case ex =>
        System.out.println(ex.getMessage());
    }

    labda(0) = 1.0 / 6.0;
    labda(1) = 1.0 / 3.0;
    labda(2) = 1.0 / 2.0;
    ngramMinimumShowTimes = Integer.parseInt(p.getProperty("NgramMinimumShowTimes", "1"));
  }

  def getNgramMinimumShowTimes = ngramMinimumShowTimes  
}
